File: cracking/arrays/Ex14.java
import java.util.HashMap;

class Ex14 {

    static boolean isPermutationOfPalindrome(String s) {
        var counters = new HashMap<Character, Integer>();
        for (char c : s.toLowerCase().toCharArray()) {
            if (Character.isLetter(c)) {
                counters.put(c, counters.getOrDefault(c, 0) + 1);
            }
        }
        var oddChars = counters.values().stream().filter(counter -> counter % 2 == 1).count();
        return oddChars < 2;
    }

    public static void main(String[] args) {
        System.out.println(isPermutationOfPalindrome("Tact Coa"));
        System.out.println(isPermutationOfPalindrome("tract"));
    }

}

File: cracking/arrays/Ex14.py

def is_palindrome_permutation(s: str) -> bool:
    counters = {}
    for c in s.lower():
        if c.isalpha():
            counters[c] = counters.get(c, 0) + 1
    odd_chars = 0
    for counter in counters.values():
        if counter % 2 == 1:
            odd_chars += 1
    return odd_chars < 2


assert is_palindrome_permutation("Tact Coa")
assert not is_palindrome_permutation("tract")
